pkg/placement/raft/state.go

```diff
@@ -126,6 +132,20 @@ func (s *DaprHostMemberState) updateAPILevel() {
 		}
 	}
 
+	// Only enforce minAPILevel if value > 0
+	// 0 is the default value of the struct.
+	// -1 is the default value of the CLI flag.
+	if s.config.minAPILevel >= uint32(0) && observedMinLevel < s.config.minAPILevel {
+		observedMinLevel = s.config.minAPILevel
+	}
+
+	// Only enforce maxAPILevel if value > 0
+	// 0 is the default value of the struct.
+	// -1 is the default value of the CLI flag.
+	if s.config.maxAPILevel >= uint32(0) && observedMinLevel > s.config.maxAPILevel {
+		observedMinLevel = s.config.maxAPILevel
+	}
```

Comment:
This looks correct to me. "maxAPILevel" is used to force a maximum value for "observedMinLevel" (which is the observed level across the cluster)

Reply:
The name of the var should probably be `minLevelObserved` instead. It is the lowest level across all members. It cannot be less than the min or greater than max.

Reply:
I confirmed that this change is required to avoid a regression where maxLevel is not enforced for reminder serialization - reproduced via an integration test.